pkg/controllermanager/controller/shoot/maintenance/reconciler.go

```diff
@@ -93,9 +97,16 @@ func requeueAfterDuration(shoot *gardencorev1beta1.Shoot) (time.Duration, time.T
 	return duration, nextMaintenance
 }
 
-func (r *Reconciler) reconcile(ctx context.Context, log logr.Logger, shoot *gardencorev1beta1.Shoot, gardenClient client.Client) error {
+func (r *Reconciler) reconcile(ctx context.Context, log logr.Logger, shoot *gardencorev1beta1.Shoot) error {
 	log.Info("Maintaining Shoot")
 
+	var (
+		now             time.Time
+		newShoot        = shoot.DeepCopy()
+		operations      []string
+		lastMaintenance = shoot.Status.LastMaintenance
```

Comment:
Will the field be present in the first reconciliation as well?

Reply:
It will be empty, but it shouldn't matter? What is your concern @shafeeqes?